python/lsst/daf/butler/script/queryDataIds.py

```diff
@@ -78,6 +82,26 @@ def queryDataIds(repo, dimensions, datasets, where, collections, order_by, limit
     # Docstring for supported parameters is the same as Registry.queryDataIds
 
     butler = Butler(repo)
+
+    if not dimensions and datasets and collections:
+        # Determine the dimensions relevant to all given dataset types.
+        graph = None
+        dataset_types = list(butler.registry.queryDatasetTypes(datasets))
+        for dataset_type in dataset_types:
+            if graph is None:
+                graph = dataset_type.dimensions
+            else:
+                graph = graph & dataset_type.dimensions
+            _LOG.debug("Dimensions now %s from %s", set(graph.names), dataset_type.name)
+            if not graph:
+                break
+
+        if not graph:
+            names = [d.name for d in dataset_types]
+            return None, f"No dimensions in common for specified dataset types ({names})"
```

Comment:
As I comment below on the other `return` statements, I think this makes more sense raising a custom exception or `NoResults`.